src/Avalonia.Controls.DataGrid/DataGrid.cs

```diff
@@ -718,6 +721,9 @@ public DataGrid()
             _mouseOverRowIndex = null;
             CurrentCellCoordinates = new DataGridCellCoordinates(-1, -1);
 
+            HorizontalScrollBar.Scroll += HorizontalScroll;
```

Comment:
I think proper place to do this is in `OnTemplateApplied`, and `DataGrid` is already subscribing to `Scroll` events there. Generally you should avoid just subscribing to event and never unsubscribing since it can easily cause memory leaks via handlers.

Reply:
Not when Scrollbar is part of DataGrid and gets disposed when DataGrid disposes. I also see no unsubscribing on KeyDown etc events either.

Reply:
Yeah, in this case it works fine - although `OnTemplateApplied` in `DataGrid` is doing exactly that.  You could probably just use `HorizontalScrollBar_Scroll` and `VerticalScrollBar_Scroll` for the same purpose and avoid double handlers.

Reply:
Thanks for the recommendation, update this PR later today

Reply:
Done